nixos/modules/services/databases/influxdb2.nix

```diff
@@ -111,58 +448,38 @@ in
           "admin-password:${cfg.provision.initialSetup.passwordFile}"
           "admin-token:${cfg.provision.initialSetup.tokenFile}"
         ];
+
+        ExecStartPost = mkIf cfg.provision.enable (
+          [provisioningScript] ++
+          # Only the restarter runs with elevated privileges
+          optional anyAuthDefined "+${restarterScript}"
```

Comment:
I've already tried realizing it using systemd alone, but I've not found a working configuration. The main issue is that RestartForceExitStatus only works when the main process exits with that status, and not pre or post scripts:

> Takes a list of exit status definitions that, when returned by the main service process [...]

Reply:
Grepped for systemctl in `nixos/modules` and found a vast number of invocations, guess that's ok. Just many of them hardcode the path to systemctl as `/run/current-system/systemd/bin/systemctl`

Reply:
And the influxdb service absorbs all signals except for SIGTERM, which I don't want to use for restarting because systemd also uses it to signal the service to exit. So if the service should be stopped, it would first be restarted and then killed after a timeout with SIGKILL. That's why I finally opted to do `systemctl restart influxdb2`, because it just was the only solution I found that doesn't mess with normal behavior.